### ISSUE TITLE
slack plugin regression (global slack plugin)

### ISSUE THREAD

**Issue by matejsp** [jenkinsci/slack-plugin#454]
I have just updated to 2.11 and global slack plugin works again! No more exceptions.
However latest version now displays additional "null" text.

Before:
![image](https://user-images.githubusercontent.com/3091122/50224264-b8ff5200-039d-11e9-995a-533ec946f27e.png)

After:
![image](https://user-images.githubusercontent.com/3091122/50224327-f19f2b80-039d-11e9-9cf9-a0adb00ac0d6.png)

**Comment by timja** [comment]
at a guess its this:
https://github.com/jenkinsci/slack-plugin/pull/448

**Comment by esproul** [comment]
I upgraded to 2.10 yesterday and notifications stopped working. I tried 2.11 this morning and still broken.  Downgrading to 2.6 works.  I'm running Jenkins 2.156.

I'm not familiar with how to troubleshoot plugin issues but I'm happy to help narrow it down if I can.

**Comment by timja** [comment]
what's not working? 2.11 should work fine other than the `null` text

**Comment by esproul** [comment]
No notifications are received in Slack.  Jenkins appears to have sent them:

```
Dec 19, 2018 11:10:42 AM jenkins.plugins.slack.SlackNotifier perform
INFO: Performing complete notifications
```

If I go into a job, under Slack Notifications, and hit Test Connection, I get a message.  But there is no notification from any job completion.  If I downgrade the plugin, with no other changes, the notifications work.

**Comment by matejsp** [comment]
It works with (with extra null text):
Global Slack Notifier Plugin 1.3
Slack Notification Plugin 2.11
Jenkins ver. 2.156

**Comment by timja** [comment]
Can you screenshot your global config please?

**Comment by esproul** [comment]
Global settings:
![global_settings](https://user-images.githubusercontent.com/731614/50227380-9f491500-0373-11e9-927f-59121dac244b.png)

Job config:
![job_config](https://user-images.githubusercontent.com/731614/50227393-a2440580-0373-11e9-854f-f147ebb3e1aa.png)

**Comment by timja** [comment]
try remove your slack compatible app url

**Comment by esproul** [comment]
Does not appear to make any difference.

**Comment by jkrehm** [comment]
I am having the same issue after updating to 2.10. Updating to 2.11 did not fix the issue, either. Test Connection works, but jobs do not trigger a message.

**Comment by timja** [comment]
@jkrehm are you using global slack notifier plugin or just slack-plugin?

**Comment by timja** [comment]
and what's the last version that works for you @jkrehm ?

**Comment by viktorkholod** [comment]
The same issue after upgrade to 2.10 and 2.11.
Jenkins 2.150.1
Slack Notification Plugin 2.11

**Comment by timja** [comment]
This issue is about `null` text when used with the global slack notifier plugin

Can you please create a new issue,
and provide your global config and job config, and the last version of the slack plugin that works for you please

**Comment by jkrehm** [comment]
@timja Just the [slack-plugin](http://wiki.jenkins-ci.org/display/JENKINS/Slack+Plugin) (I was not aware of the global plugin until this thread).

I upgraded to 2.10 and then 2.11 (hoping it would fix the issue) so I am not able to drop back beyond that, but based on configuration backups, I believe 2.3 was what was in use. I'm guessing that won't be much help diagnosing the issue because of how far back it is.

**Comment by timja** [comment]
unfortunately not since 2.3 was over a year ago and lots of changes have been merged since.
does your issue look like this one?
https://github.com/jenkinsci/slack-plugin/issues/455

I just posted all my config there (which is working for me...) i've been trying to reproduce these issues but not sure what is causing it

**Comment by jkrehm** [comment]
Since my issue was with slack-plugin (not the global slack-plugin) #457 fixed the issue for me. Please ignore my comments above.

**Comment by sergeykranga** [comment]
I had the same issue: upgraded to Jenkins version 2.150.1, slack plugin version 2.11
We use Job DSL to describe/migrate jobs and CasC to configure Slack when newer Jenkins versions appear, so this was not the issue with how jobs/slack was configured.

We never used "Notify regression" option, but I noticed that when you enable it, it starts sending the notifications again. With it disabled, any other combination of options didn't work for me. I looked into the source, and didn't find anything related to this (or maybe I looked in a wrong place). System log also showed no errors.

Solution that worked for me is to upgrade Slack plugin to 2.12, which just recently appeared in plugin manager
